hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -160,10 +180,61 @@ func createFailureConditionToNodePoolsCountMap() *map[string]int {
 	return &res
 }
 
+func (c *nodePoolsMetricsCollector) resolveCoresCountPerNode(nodePool *hyperv1.NodePool, unresolvedEc2InstanceTypes *map[string]void) int32 {
+	if nodePool.Spec.Platform.Type == hyperv1.AWSPlatform && c.ec2Client != nil {
```

Comment:
can we instead return early if this is not true instead of having the whole logic within the if statement?

Reply:
We don't need to handle multiple platforms atm. If we ever need that in the future then will re-evaluate and accommodate the code for it.
For now please let's return earlier as that's the most readable approach for this particular piece of code